File: programers_Lv1/문자열 내림차순으로 배치하기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/12917?language=javascript
function solution(s) {
  alphaLower = [..."abcdefghijklmnopqrstuvwxyz"];

  console.log(alphaLower);

  const sArr = [...s];

  const first = [];
  const second = [];

  sArr.forEach((el) => {
    if (alphaLower.includes(el)) {
      first.push(el);
    } else {
      second.push(el);
    }
  });

  return first.sort().reverse().join("") + second.sort().reverse().join("");
}

console.log(solution("Zbcdefg"));

File: programers_Lv1/문자열 내림차순으로 배치하기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12917?language=python3
import string
def solution(s):
    alpha_lower = string.ascii_lowercase

    first = []
    second = []

    for i in s:
        if i in alpha_lower:
            first.append(i)
        else:
            second.append(i)

    return ''.join(sorted(first, reverse=True)) + ''.join(sorted(second, reverse=True))

print(solution("Zbcdefg"))
